src/main/java/wellnus/storage/Tokenizer.java

```diff
@@ -0,0 +1,37 @@
+package wellnus.storage;
+
+import wellnus.exception.TokenizerException;
+import wellnus.manager.Manager;
+
+import java.util.ArrayList;
+
+/**
+ * Template for all Tokenizers in WellNUS++ that are responsible for converting
+ *     Managers into Strings(for storage) and also Strings(from storage) back
+ *     into Managers with the previously saved state.<br/>
+ *
+ * Example of how to implement this in a feature: <code>public class AtomicHabitTokenizer
+ *     implements Tokenizer&lt;AtomicHabit&gt;</code>.
+ * @param <T> Data type of the corresponding feature, e.g. <code>AtomicHabit</code>
+ *     the atomic habit feature
+ */
+public interface Tokenizer<T> {
```

Comment:
The <T> generic here might not be necessary!

The data structure that is to be tokenized is encapsulated in Manager already. Since the Manager is a parameter in the function signature, I don't think there is a need to force Tokenizer to inherit a data type/manager on implementation